agent-lib/src/main/java/org/terracotta/angela/agent/com/IgniteFutureAdapter.java

```diff
@@ -53,6 +62,9 @@ public boolean isDone() {
   public V get() throws InterruptedException, ExecutionException {
     try {
       return igniteFuture.get();
+    } catch (ClusterGroupEmptyException cge) {
+      logger.warn("DETECTED POTENTIAL UNEXPECTED FAILURE (OR KILL) OF JVM WITH NODE: {}", agentID);
+      throw new IllegalStateException("Agent is gone in an abrupt way and Ignite cannot get any result from it anymore: " + agentID, cge);
     } catch (IgniteInterruptedException iie) {
       throw (InterruptedException) new InterruptedException().initCause(iie);
     } catch (IgniteException ie) {
```

Comment:
can't we check here if IgniteException  instanceOf ClusterGroupEmptyException

Reply:
why ?

usually you catch the more specific exception first and the one with higher hierarchy after...

Reply:
okay, right now in my perf test IgniteException is getting thrown so thought of more specific check there.
but anyways ClusterGroupEmptyException is already covered.